### ISSUE TITLE
whitelist does not work with '>'

### ISSUE THREAD

**Issue by lager1** [loris/api-query-params#78]
```
const aqp = require('api-query-params').default; 
const test = aqp('timestamp>2016-01-01');
 console.log(test);
 const test2 = aqp('timestamp>2016-01-01',{
   whitelist : ['timestamp']
 });
 console.log(test2);
```
the code above gives : 

```
{ filter: { timestamp: { '$gt': 2016-01-01T00:00:00.000Z } } }
{ filter: {} }
```

```
$ node -v
v6.5.0
```

**Comment by lager1** [comment]
The code below should fix this issue.

```
function getFilter(filter, query, options) {
  if (filter) {
    try {
      return JSON.parse(filter);
    } catch (err) {
      throw new Error(`Invalid JSON string: ${filter}`);
    }
  }

  return Object.keys(query)
    .reduce((result, val) => {
      const join = query[val] ? `${val}=${query[val]}` : val;
      // Separate key, operators and value
      const [, prefix, key, _op, _value] = join.match(/(!?)([^><!=]+)([><]=?|!?=|)(.*)/);

      const value = parseValue(_value);
      const op = parseOperator(_op);

      if (!result[key]) {
        result[key] = {};
      }

      if (Array.isArray(value)) {
        result[key][op === '$ne' ? '$nin' : '$in'] = value;
      } else if (op === '$exists') {
        result[key][op] = prefix !== '!';
      } else if (op === '$eq') {
        result[key] = value;
      } else if (op === '$ne' && typeof value === 'object') {
        result[key].$not = value;
      } else {
        result[key][op] = value;
      }

      const result_keys = Object.keys(result);

      for(var item in result_keys) { // iterate all defined filter keys
        if(!(options.blacklist.indexOf(result_keys[item]) === -1  // does not exist in blacklist
        && (!options.whitelist || options.whitelist.indexOf(result_keys[item]) !== -1))) // and can be defined in whitelist
          delete result[result_keys[item]]; // delete item from result
      }
      return result;
    }, {});
}
```

**Comment by loris** [comment]
Thank you for reporting, it is fixed in `4.0.1`
